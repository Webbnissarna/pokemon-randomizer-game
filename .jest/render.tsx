import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
