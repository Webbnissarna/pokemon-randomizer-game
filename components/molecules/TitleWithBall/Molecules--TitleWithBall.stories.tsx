import TitleWithBall from ".";

export function Default() {
  return <TitleWithBall text="PokÃ©mon Randomizer Game!" />;
}
