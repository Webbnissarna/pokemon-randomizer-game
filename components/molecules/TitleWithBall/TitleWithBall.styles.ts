import { ThemeUIStyleObject } from "theme-ui";

export const rootStyle: ThemeUIStyleObject = {
  position: "relative",
  alignItems: "center",
  justifyContent: "center",
  height: 250,
};

export const ballContainerStyle: ThemeUIStyleObject = {
  width: 250,
  height: 250,
  position: "absolute",
  top: 0,
  zIndex: 0,
  left: "calc(50% - 125px)",
};

export const titleContainerStyle: ThemeUIStyleObject = {
  zIndex: 1,
  position: "relative",
  width: "100%",
};

export const titleStyle: ThemeUIStyleObject = {
  textAlign: "center",
};
