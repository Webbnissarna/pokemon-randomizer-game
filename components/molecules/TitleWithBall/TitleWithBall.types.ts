export interface TitleWithBallProps {
  text: string;
}
