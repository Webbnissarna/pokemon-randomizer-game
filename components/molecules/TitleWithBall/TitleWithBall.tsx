import { Box, Flex, Heading } from "theme-ui";
import CardBox from "../../atoms/CardBox";
import SimpleBall from "../../atoms/SimpleBall";
import {
  rootStyle,
  ballContainerStyle,
  titleContainerStyle,
  titleStyle,
} from "./TitleWithBall.styles";
import { TitleWithBallProps } from "./TitleWithBall.types";

export default function TitleWithBall({
  text,
}: TitleWithBallProps): JSX.Element {
  return (
    <Flex sx={rootStyle}>
      <Box sx={ballContainerStyle}>
        <SimpleBall size={"100%"} />
      </Box>
      <Box sx={titleContainerStyle}>
        <CardBox>
          <Heading as="h1" sx={titleStyle}>
            {text}
          </Heading>
        </CardBox>
      </Box>
    </Flex>
  );
}
