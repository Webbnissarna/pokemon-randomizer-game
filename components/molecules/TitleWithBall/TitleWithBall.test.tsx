import { render } from "../../../.jest/render";
import TitleWithBall from ".";

describe("TitleWithBall", () => {
  it("renders", () => {
    const { asFragment } = render(<TitleWithBall text="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
