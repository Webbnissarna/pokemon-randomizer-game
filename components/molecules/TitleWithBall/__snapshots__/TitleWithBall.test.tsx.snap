// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithBall renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-1vj68q-Box"
  >
    <div
      class="css-x01a4x-Box"
    >
      <svg
        height="100%"
        style="filter: drop-shadow(rgba(0, 0, 0, 0.25) 0px 4px 4px);"
        version="1.1"
        viewBox="0 0 200 200"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width=".7"
          transform="translate(-4.612 -47.502)"
        >
          <path
            d="m104.61 47.502a100 100 0 0 0-100 100h200a100 100 0 0 0-100-100z"
            fill="var(--theme-ui-colors-ball-active-top)"
            stop-color="#000000"
          />
          <path
            d="m104.61 247.5a100 100 0 0 0 100-100h-200a100 100 0 0 0 100 100z"
            fill="var(--theme-ui-colors-ball-active-bottom)"
            stop-color="#000000"
          />
        </g>
      </svg>
    </div>
    <div
      class="css-jgcf2z-Box"
    >
      <div
        class="css-wimkxn-Box"
      >
        <h1
          class="css-1kdz7xw-Box"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
